content/api/commands/session.md

```diff
@@ -135,14 +134,14 @@ The page and all active session data (cookies, `localStorage` and
-| ---------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| `validate` | `undefined` | Validates the newly-created or restored session.<br><br>The `validate` function is run immediately after the `setup` function runs, and also every time `cy.session()` restores a cached session. If the `validate` function returns `false`, throws an exception, returns a Promise that resolves to `false` or rejects, or contains any failing Cypress command, the session will be considered invalid, and `setup` will be re-run. If validation fails immediately after `setup` runs, the test will fail.<br><br>The page is always cleared before `validate` runs. |
+| Option             | Default     | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
+| ------------------ | ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `validate`         | `undefined` | Validates the newly-created or restored session.<br><br>Function to run immediately after the session is created and `setup` function runs or after a session is restored and the page is cleared. If this returns `false`, throws an exception, returns a Promise which resolves to `false` or rejects or contains any failing Cypress command, the session is considered invalid.<br><br>- If validation fails immediately after `setup`, the test will fail.<br>- If validation fails after restoring a session, `setup` will re-run. |
```

Comment:
This section reads a bit weird to me. If `session` returns false, does an exception get thrown in addition to a promise being returned which resolves to `false` and/or rejects with a failing command?

"If this returns `false`, throws an exception, returns a Promise which resolves to `false` or rejects or contains any failing Cypress command"

Reply:
These are examples of validate functions that will fail validation:
```js
validate = () => {
	expect(true).to.be.false // failed assertion throws (or something else that throws)
}

validate = () => {
	cy.get('#does_not_exist') // fails command
}

validate = () => {
	return false
}

validate = async () => {
	return Promise.resolve('false')
}

validate = async () => {
	return Promise.reject()
}
```

Reply:
how about this? i slightly reordered the sentence. 

```suggestion
| `validate`         | `undefined` | Validates the newly-created or restored session.<br><br>Function to run immediately after the session is created and `setup` function runs or after a session is restored and the page is cleared. If this returns `false`, throws an exception, contains any failing Cypress command, or returns a Promise which rejects or resolves to `false`, the session is considered invalid.<br><br>- If validation fails immediately after `setup`, the test will fail.<br>- If validation fails after restoring a session, `setup` will re-run. |
```